--- net/sched/sch_netem.c	2014-01-19 20:40:07.000000000 -0600
+++ /home/dslab/sch_netem_joseph.c	2015-04-02 19:40:05.842182702 -0500
@@ -559,8 +559,9 @@ static struct sk_buff *netem_dequeue(str
 tfifo_dequeue:
 	skb = __skb_dequeue(&sch->q);
 	if (skb) {
-deliver:
+		/* jbeshay: update byte backlog to match queued packet count */
 		sch->qstats.backlog -= qdisc_pkt_len(skb);
+deliver:
 		qdisc_unthrottled(sch);
 		qdisc_bstats_update(sch, skb);
 		return skb;
@@ -577,6 +578,8 @@ deliver:
 			rb_erase(p, &q->t_root);
 
 			sch->q.qlen--;
+			sch->qstats.backlog -= qdisc_pkt_len(skb);
+
 			skb->next = NULL;
 			skb->prev = NULL;
 			skb->tstamp = netem_skb_cb(skb)->tstamp_save;
